lib/Monitor.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace Cronitor;
+
+class Monitor
+{
+    private const BASE_MONITOR_API_URL = 'https://cronitor.io/api/monitors';
+    private const BASE_PING_API_URL = "https://cronitor.link/p";
+    private const BASE_FALLBACK_PING_URL = "https://cronitor.link/p";
+    private const PING_RETRY_THRESHOLD = 5;
+
+    public $apiKey;
+    public $key;
+
+    public function __construct($apiKey, $key)
+    {
+        $this->apiKey = $apiKey;
+        $this->key = $key;
+
+        $monitorApiUrl = self::BASE_MONITOR_API_URL . "/$key";
+        $this->monitorClient = new HttpClient($monitorApiUrl, $this->apiKey);
+    }
+
+    public static function put($apiKey, $params = [])
+    {
+        $rollback = $params['rollback'] || false;
+        unset($params['rollback']);
+        $monitors = $params['monitors'] || [$params];
+
+        $client = new HttpClient(self::BASE_MONITOR_API_URL, $apiKey);
+        $response = $client->put('', [
+            'monitors' => $monitors,
+            'rollback' => $rollback,
+        ], ['timeout' => 10]);
+
+        $code = $response['code'];
+        switch ($code) {
+            case 200:
+                $out = [];
+                $data = json_decode($response['content']);
+
+                foreach ($data['monitors'] || [] as &$md) {
+                    $m = new Monitor($md['key']);
+                    $m->data = $md;
+                    array_push($out, $m);
+                }
+                return count($out) == 1 ? $out[0] : $out;
+                break;
+            case 400:
+                throw new ValidationException($response['content']);
+            default:
+                throw new \Exception("Error connecting to Cronitor: $code");
+        }
+    }
+
+    public static function delete($key, $apiKey)
+    {
+        $client = new HttpClient(self::BASE_MONITOR_API_URL . "/$key", $apiKey);
+        $response = $client->delete($path, ['timeout' => 10]);
+        
+        if ($response['code'] != 204) {
+            \error_log("Error deleting monitor: $key", 0);
+            return false;
+        }
+        return $response;
+    }
+
+    public function ping($params = array())
+    {
+        $retryCount = $params['retryCount'] || 0;
+
+        if (!$this->apiKey) {
+            \error_log('No API key detected. Set Cronitor.api_key or initialize Monitor with an api_key:', 0);
+            return false;
+        }
+
+        try {
+            $queryString = $this->buildPingQuery($params);
+            $client = $this->getPingClient($retryCount);
+            $response = $client->get("/?$queryString");
+            $responseCode = $response['code'];
+
+            if ($responseCode < 200 || $responseCode > 299) {
```

Comment:
The Ruby library was checking a straight 200 response, but I see a 204 code. Regardless, checking against the full 200-range seems to be the best practice.